nodedb.go

```diff
@@ -116,17 +116,22 @@ func (ndb *nodeDB) SaveNode(node *Node) {
 }
 
 // Has checks if a hash exists in the database.
-func (ndb *nodeDB) Has(hash []byte) bool {
+func (ndb *nodeDB) Has(hash []byte) (bool, error) {
 	key := ndb.nodeKey(hash)
 
 	if ldb, ok := ndb.db.(*dbm.GoLevelDB); ok {
 		exists, err := ldb.DB().Has(key, nil)
 		if err != nil {
-			panic("Got error from leveldb: " + err.Error())
```

Comment:
do we want to keep the panic here? or let the end-user handle it? do you have an opinion @alexanderbez

Reply:
It is of my opinion a DB library should never panic -- allow the upstream user to decide on how to handle the error.

Reply:
Strongly endorse ^